Feature/week8 homework

Bartozo opened the pull request:
My finished homework:

-I added a simple unique worker which downloads the data from the api
-downloaded data is stored in the database

Bartozo committed:
Added work library

Bartozo committed:
Changed library for worker and Added empty worker class

- I have changed library for worker to be able to use kotlin and coroutines

Bartozo committed:
Added a Worker class

Bartozo committed:
Added worker usage in application class

- worker is unique = only 1 instance of it
- it downloads the data from the api and store it in the database

filbabic commented on Week8/app/src/main/java/com/bartoszkrol/catfacts/App.kt:
```diff
@@ -17,6 +21,31 @@ class App : Application() {
+            .setConstraints(constraints)
+            .build()
+
+        val workManager = WorkManager.getInstance(this)
```
Really nice and well structured! :]

However, as someone (may be you) mentioned, the minimum period is 15 minutes :/

But I'll take it as if it's working, so don't worry.

Bartozo replied:
Yes, the minimum period is 15minutes (I think it apply for JobScheduler too).

I forgot to change repeatInterval value and timeUnit. I was testing it on the real device to see if its gonna work every 1 minute.

filbabic commented on Week8/app/src/main/java/com/bartoszkrol/catfacts/worker/DownloadDataWorker.kt:
```diff
@@ -0,0 +1,35 @@
+        if (result is Success) {
+            // Add new data to the database
+            // LiveData will update recyclerView with a new facts about the cats
+            val catFactsDao = CatFactsDatabase.getDatabase(applicationContext).catFactsDao()
```
It would be better if you had factory functions for these things, so you don't have to do this manually here!

Bartozo replied:
Now dao and repository are available from the app class.

filbabic reviewed (approved):
Not much to say, just a small comment as per the previous assignment.

**Good Job** on the assignment!

Bartozo committed:
Added fixes from the previous week

-missing app in manifest
- repository and dao are now in the app class
- added viewModelFactory for CatFactsViewModel

Bartozo committed:
Fixed the code

- removed initialised dao and repository inside DownloadDataWorker class
- changed worker time unit to Hours (now it gonna download new data every 1 hour)
